### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by Ilanak**

### CODE REVIEW

File: test/awsS3ModuleTest.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      Bucket: bucketName,
+      Delimiter: "/",
+    },
+  });
+
+  it("downloadFileFromS3 downloads the file to local folder successfully", async (done) => {
+    // given
+    createDirIfNotExists("./", bucketName);
+    createDirIfNotExists("./", tempFolder);
+    fs.appendFile(awsFileTemp, fileContent, function (err) {
```
Comment (morsh) [inline test/awsS3ModuleTest.ts]: `err => {`

File: test/awsS3ModuleTest.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  let s3 = AWSMock.S3({
+    params: {
+      Bucket: bucketName,
+      Delimiter: "/",
+    },
+  });
+
+  it("downloadFileFromS3 downloads the file to local folder successfully", async (done) => {
+    // given
+    createDirIfNotExists("./", bucketName);
```
Comment (morsh) [inline test/awsS3ModuleTest.ts]: " ==> '

File: test/awsS3ModuleTest.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      expect(err).to.equal(null);
+    });
+
+    // act
+    awsS3Module = new AwsS3Module(bucketName, tempFolder, s3);
+    let result = await awsS3Module.downloadFileFromS3({ Key: "newfile.txt" });
+    
+    // verify file is downloaded and it"s content
+    fs.readFile(newFilePath, "utf8", function (err, data) {
+      expect(fileContent).to.deep.equal(data);
```
Comment (morsh) [inline test/awsS3ModuleTest.ts]: check err for errors, and also do try/catch, if there an error in deleteFile/removeDir, done will not be called

File: test/azureDataLakeModuleTest.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import * as msrestAzure from "ms-rest-azure";
+import * as sinon from "sinon";
+import { AzureDataLakeModule } from "../src/azureDataLakeModule";
+
+describe("shouldUploadToADL tests", () => {
+    let adlClient: adlsManagement.DataLakeStoreFileSystemClient;
+    let adlModule: AzureDataLakeModule;
+
+    beforeEach(function () {
+        const credentials = new msrestAzure.ApplicationTokenCredentials("someclient", "domain", "secret");
```
Comment (morsh) [inline test/azureDataLakeModuleTest.ts]: Is this a live test or a mock?
Not sure since I don't see any mock files

Comment (Ilanak) [inline test/azureDataLakeModuleTest.ts]: i'm using sinon for mocking..

File: src/logger.ts [hunk 5]
```diff
@@ -1,15 +1,15 @@
-import { Logger, transports } from 'winston';
+import { transports, Logger } from "winston";
 
 export let winston = new Logger({
   transports: [
     new transports.Console(),
-    new transports.File ({ filename: process.env.TEMP_FOLDER +'/logfile.log' })
-  ]
+    new transports.File({ filename: process.env.TEMP_FOLDER + "/logfile.log" }),
```
Comment (Ilanak) [inline src/logger.ts]: I already doing it in the ctor of s3ToAdlDataFactory -  but I'll add a default value..

**Pull request merged**